src/expipe_plugin_cinpla/tools/track_units_tools.py

```diff
@@ -6,58 +6,32 @@
 from scipy.optimize import linear_sum_assignment
 
 
-def dissimilarity(template_0, template_1):
+def compute_dissimilarity(template_0, template_1):
     """
-    Returns a value of dissimilarity of the mean between two or more
-    spike templates.
-    Parameters
-    ----------
-    templates : list object (see Notes)
-        List containing the mean waveform over each electrode of spike sorted
-        spiketrains from at least one electrode. All elements in the list must
-        be of equal size, that is, the number of electrodes must be equal, and
-        the number of points on the waveform must be equal.
-    Returns
-    -------
-    diss : numpy array-like
-        Returns a matrix containing the computed dissimilarity between the mean
-        of the spiketrain, for the same channel.
-    """
-    max_val = np.max([np.max(np.abs(template_0)), np.max(np.abs(template_1))])
-
-    t_i_lin = template_0.ravel()
-    t_j_lin = template_1.ravel()
-
-    return np.mean(np.abs(t_i_lin / max_val - t_j_lin / max_val))
-    # return np.mean(np.abs(t_i_lin - t_j_lin))
-
+    Returns a value of dissimilarity of the mean between two templates.
+    The dissimilarity is computed as the root sum square of the difference
+    between the two templates, averaged over channels and normalized by the
+    maximum value of the two templates.
 
-def dissimilarity_weighted(templates_0, templates_1):
-    """
-    Returns a value of dissimilarity of the mean between two or more
-    spike templates.
     Parameters
     ----------
-    templates : list object (see Notes)
-        List containing the mean waveform over each electrode of spike sorted
-        spiketrains from at least one electrode. All elements in the list must
-        be of equal size, that is, the number of electrodes must be equal, and
-        the number of points on the waveform must be equal.
+    template_0 : np.ndarray
+        The first template. Dimensions are (n_timepoints, n_channels).
+    template_1 : np.ndarray
+        The second template. Dimensions are (n_timepoints, n_channels).
+
     Returns
     -------
-    diss : numpy array-like
-        Returns a matrix containing the computed dissimilarity between the mean
-        of the spiketrain, for the same channel.
+    dissimilarity : float
+        The dissimilarity between the two templates.
     """
 
-    max_val = np.max([np.max(np.abs(templates_0)), np.max(np.abs(templates_1))])
+    max_val = np.max([np.max(np.abs(template_0)), np.max(np.abs(template_1))])
 
-    templates_0 /= max_val
-    templates_1 /= max_val
+    template_0_scaled = template_0 / max_val
+    template_1_scaled = template_1 / max_val
```

Comment:
This was the main issue. In the previous version, this function took waveforms and set the templates as their means. With the refactor, this got lost and the problem is that th `\=` operator changes the variable in memory. So calling this function multiple times was increasingly making the templates smaller...very hard to find!